include/swift/AST/DiagnosticsSema.def

```diff
@@ -966,6 +966,8 @@ ERROR(extra_address_of,none,
 ERROR(extra_address_of_unsafepointer,none,
       "'&' is not allowed passing array value as %0 argument",
       (Type))
+ERROR(cannot_pass_inout_arg_to_subscript,none,
+      "cannot pass an inout argument to a subscript", ())
```

Comment:
@jrose-apple What do you think about this diagnostic message?

Reply:
Also going to add TODO to the test so it's clear that there is a possible improvement...

Reply:
Oh, I didn't even mean doing it *well.* Thinking about it a little more, the syntactic check can say "this is an inout argument in subscript index position; let's just always suggest `withUnsafeMutablePointer`". I wasn't even thinking you'd include a fix-it, although obviously that would be nice.

Reply:
Ah you mean - `cannot pass an inout argument to a subscript; use 'withUnsafeMutablePointer' to explicitly convert argument to a pointer`?

Reply:
Yeah, something like that.

Reply:
Okay, I'll add that momentarily!